src/gui/guiFormSpecMenu.cpp

```diff
@@ -2707,6 +2872,66 @@ void GUIFormSpecMenu::regenerateGui(v2u32 screensize)
 		setInitialFocus();
 
 	skin->setFont(old_font);
+
+	// legacy sorting
+	if (m_formspec_version < 2)
+		legacySortElements(legacy_sort_start++);
+}
+
+void GUIFormSpecMenu::legacySortElements(core::list<IGUIElement *>::Iterator from)
+{
+	/*
+	 * The order was:
+	 * (0. background colors and backgrounds)
+	 * 1. boxes
+	 * 2. all normal elements (like buttons)
+	 * 3. images
+	 * 4. item images
+	 * 5. item lists (TODO)
```

Comment:
Is this needed given that the new system draws inv lists in front anyway? This would explain why this is todo

Reply:
Yes, the lists are still drawn in front with this PR. But this should be fixed at some point, hence the todo.